test/Conversion/StandardToHandshake/test_remove_block.mlir

```diff
@@ -0,0 +1,31 @@
+// NOTE: Assertions have been autogenerated by utils/update_mlir_test_checks.py
+// RUN: circt-opt -create-dataflow -remove-block-structure %s | FileCheck %s
+
+// CHECK: module {
+// CHECK-LABEL: handshake.func @dfs_block_order(
+// CHECK-SAME:  %[[VAL_0:.*]]: none, ...) -> (i32, none) {
+func @dfs_block_order() -> (i32) {
+
+  // CHECK: handshake.constant
+  %0 = constant 42 : i32
+  // CHECK-NOT: "handshake.terminator"()[^bb2] : () -> ()
```

Comment:
you could also get the same effect using "CHECK-NEXT" instead of "CHECK" to ensure the terminators are not present.

Reply:
There will be some other operations (e.g. merge, branch) between them, thus "CHECK-NEXT" may not work here.

Reply:
If the startpoint is the dataflow IR, this should work.